uniquify.go

```diff
@@ -0,0 +1,70 @@
+package frontman
+
+import (
+	"fmt"
+	"sync"
+)
+
+// Uniquify is a type of advanced mutex. It allows to create named resource locks.
+type Uniquify interface {
+	// Call executes only one callable with same id at a time.
+	// Multilpe asynchronous calls with same id will be executed sequentally.
+	Call(id string, callable func() error) error
+}
+
+// NewUniquify returns a new thread-safe uniquify object.
+func NewUniquify() Uniquify {
+	return &uniquify{
+		tasks: make(map[string]*sync.WaitGroup),
+	}
+}
+
+type uniquify struct {
+	lock  sync.Mutex
+	tasks map[string]*sync.WaitGroup
+}
+
+func (u *uniquify) Call(id string, callable func() error) error {
```

Comment:
I suggest to add some comments to this non-trivial code.

Reply:
The function is commented in the interface above, this is an implementation.

Reply:
And if there is a bug or problem or similar in the future then there is no need to understand the implementation?

Reply:
The comments are needed for understanding the API and use cases. To avoid misuse.

If person wants to learn the implementation, he should read the code, especially if he wants to modify it. You cannot extend existing code by just reading comments.

Reply:
This functions needs an explanation comment. Please do not confuse API documentation with implementation